core/storage/src/storages/shard.rs

```diff
@@ -21,6 +21,8 @@ pub struct ShardChainStorage {
     transaction_results: HashMap<Vec<u8>, TransactionResult>,
     transaction_addresses: HashMap<Vec<u8>, TransactionAddress>,
     receipts: HashMap<Vec<u8>, HashMap<ShardId, ReceiptBlock>>,
+    // records the largest transaction nonce in a given block
+    tx_nonce: HashMap<Vec<u8>, u64>,
```

Comment:
As explained in the other comment, I think we should be storing largest nonce per sender per block, instead of largest nonce per block.